### PR TITLE
BREAKING CHANGE: remove legacy `detectron2` model; remove layoutparser extras

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

First step in resolving https://github.com/Unstructured-IO/unstructured/issues/3051. Per [this comment](https://github.com/Unstructured-IO/unstructured/issues/3051#issuecomment-2123176161), we were having troubling running `unstructured` in the Python 3.12 `wolfi-base` contain due to issues related to `pycocotools`, which is only used for the legacy `detectron2` model from `layoutparser`. Since we've replaced this with `detectron2onnx`, this PR removes the `layoutparser` extra dependencies that caused issues with Python 3.12.

The `layoutparser` base dependency is still required because we use layout objects from that library. It's likely we could remove these in a future iteration.

Temporarily disabled the ingest tests, because they seem to have been broken for the past six months. Last commit that they passed for was [this one](https://github.com/Unstructured-IO/unstructured-inference/commit/0f0c2be07c673615217b214937f50a091676a8aa). Opened #352 to reenable them.

### Testing

If CI passes we should be good to go.

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -109,24 +99,18 @@ pillow==10.3.0
...
     #   pdfplumber
-    #   pytesseract
-    #   torchvision
 portalocker==2.8.2
     # via iopath
 protobuf==5.26.1
     # via
     #   onnx
     #   onnxruntime
-pycocotools==2.0.7
```
Comment (MthwRobinson) [inline requirements/base.txt:120]: Here's the dependency we needed to remove

File: requirements/base.in [hunk 2]
```diff
@@ -1,10 +1,12 @@
 -c constraints.in
-layoutparser[layoutmodels,tesseract]
+layoutparser
```
Comment (qued) [inline requirements/base.in:2]: Are we not able to lose `layoutparser` completely?

Comment (MthwRobinson) [reply requirements/base.in:2]: We still have this `TextBox` element we rely on from `layoutparser`, but I wonder if we can just copy that over

https://github.com/Unstructured-IO/unstructured-inference/blob/81549a7e24b66089995bfe7c619f35ee6774bda3/unstructured_inference/inference/layoutelement.py#L7

**Comment by MthwRobinson** [comment]
@qued - Tests are passing now, should be good for final review!

**Review by qued — APPROVED** [review]
LGTM!

**Pull request merged**